exercicios/para-casa/entregas/kauana-tombolato/Personagem.js

```diff
@@ -0,0 +1,39 @@
+import { SuperPoder } from './SuperPoder.js'
+
+export class Personagem {
+  #nome
+  #nomeVidaReal
+  #poderes
+
+  constructor(nome, nomeVidaReal, poderes) {
+    this.#nome = nome
+    this.#nomeVidaReal = nomeVidaReal
+    this.#poderes = poderes || []
```

Comment:
Aqui uma pergunta de curiosidade mesmo: como funciona essa linha? "this.#poderes = poderes" eu entendi, mas o "| | [ ]" ficou nebuloso pra mim.

Reply:
Olá Andrea, desculpa demora para responder, mas funciona da seguinte forma, quando eu for instanciar a classe Personagem fazendo dessa forma eu consigo já adicionar poderes.
ex: const personagem1 = new Personagem("Superman", "Clark Kent", [superPoder1, superPoder2])

Assim como, consigo também criar um Personagem sem poderes algum.
ex: const personagem1 = new Personagem("Superman", "Clark Kent")
Posteriormente teria que fazer uso da função para adicionar poderes.

Nesse caso o construtor recebe poderes de alguma forma, pois a condição é que poderes = poderes(que será definido quando instanciado) ou [].
 
Nesse último caso a instancia entende que será uma array vazia, pois não foi definido.

Espero que você tenha entendido, caso não pode me chamar 😃